core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3045,6 +3045,24 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleAlterIsrRequest(request: RequestChannel.Request): Unit = {
+    val alterIsrRequest = request.body[AlterIsrRequest]
+
+    if (!controller.isActive) {
```

Comment:
I think authorization should probably be the first thing we do.

Reply:
This is actually a really good point. I filed a JIRA to fix this in other places in KafkaApis https://issues.apache.org/jira/browse/KAFKA-10491